synapse/handlers/presence.py

```diff
@@ -1255,9 +1250,10 @@ async def set_state(
         if not ignore_status_msg:
             new_fields["status_msg"] = status_msg
 
-        if presence == PresenceState.ONLINE or (
-            presence == PresenceState.BUSY and self._busy_presence_enabled
-        ):
+        if (
+            prev_state.state != PresenceState.ONLINE
+            and presence == PresenceState.ONLINE
+        ) or (presence == PresenceState.BUSY and self._busy_presence_enabled):
```

Comment:
This was the meat and potatoes of the fix, if we are already `online`, we don't have to bump `last_active_ts`